src/schema/schema_validators.py

```diff
@@ -509,6 +509,27 @@ def validate_publication_date(property_key, normalized_entity_type, request, exi
     except ValueError:
         raise ValueError(f"Invalid {property_key} format, must be YYYY-MM-DD")
 
+"""
+Validate the provided value of the activity creation action. Only very specific
+values are allowed.
+Parameters
+----------
+property_key : str
+    The target property key
+normalized_type : str
+    Submission
+request: Flask request object
+    The instance of Flask request passed in from application request
+existing_data_dict : dict
+    A dictionary that contains all existing entity properties
+new_data_dict : dict
+    The json data in request body, already after the regular validations
+"""
+def validate_creation_action(property_key, normalized_entity_type, request, existing_data_dict, new_data_dict):
+    creation_action = new_data_dict[property_key]
+    if creation_action and creation_action not in ["Central Process", "Lab Process", "Multi-Assay Split"]:
```

Comment:
I would make this string match case-insensitive so it's more fault-tolerant, just like the way we check `status` values. It's also a good idea to tell the API consumers what are the accepted values in the `ValueError()` next line. Because these values are not as easy-to-get-right-for-the-first-time as those simple status values.

Reply:
@yuanzhou I changed `new_data_dict['creatoin_action']` to `new_data_dict.get('creation_action') which on its own should fix the error being raised. While I think it might be acceptable to just use the default creation action when the field is empty, I still went ahead and added validation to make sure its not empty anyway. I made the validation case insensitive and added some case normalization within the trigger so it will always display with the first letters capitalized (e.g. "Lab Process"). I modified the value error message to include the accepted values for creation_action as well.